go/checks/checks.go

```diff
@@ -178,4 +178,12 @@ func CheckMaxPartitionsContributed(label string, maxPartitionsContributed int64)
 		return fmt.Errorf("%s: MaxPartitionsContributed is %d, should not be negative", label, maxPartitionsContributed)
 	}
 	return nil
-}
\ No newline at end of file
+}
+
+// CheckConfidenceLevel returns an error if the supplied confidence level is not between 0 and 1
+func CheckConfidenceLevel(label string, confidenceLevel float64) error {
+	if confidenceLevel < 0 || confidenceLevel > 1 || math.IsNaN(confidenceLevel) {
```

Comment:
should we probably check if it is Infinity?

then change the text in the braces "(and cannot be infinity or NaN)" for consistency with other checks.